rag/infrastructure/mongo/pagination_builder.py

```
@@ -0,0 +1,262 @@
+
+        if distinct_field:
+            # ─── DISTINCT VALUES MODE ─────────────────────────────────────
+            pipeline.append({"$unwind": f"${distinct_field}"})
+            
+            search_match = self._build_search_match(distinct_field)
+            if search_match:
+                pipeline.append({"$match": search_match})
+            else:
+                # Filter out null/empty values
+                pipeline.append({"$match": {
+                    distinct_field: {"$exists": True, "$ne": None, "$ne": ""}
```

Ruff (0.14.13)

Dictionary key literal `"$ne"` repeated

(F601)